File: cpp/Ice/locator/HelloI.cpp
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

#include "HelloI.h"
#include <Ice/Ice.h>
#include <iostream>

using namespace std;

void
HelloI::sayHello(const Ice::Current&)
{
    cout << "Hello World!" << endl;
}

void
HelloI::shutdown(const Ice::Current& c)
{
    cout << "Shutting down..." << endl;

    //
    // Unregister from the Locator registry
    //
    auto communicator = c.adapter->getCommunicator();
    communicator->getDefaultLocator()->getRegistry()->setAdapterDirectProxy("Hello", nullopt);
    communicator->shutdown();
}

File: java-compat/Ice/plugin/HelloI.java
// **********************************************************************
//
// Copyright (c) 2003-present ZeroC, Inc. All rights reserved.
//
// **********************************************************************

import Demo.*;

public class HelloI extends _HelloDisp
{
    @Override
    public void
    sayHello(Ice.Current current)
    {
        current.adapter.getCommunicator().getLogger().print("Hello World!");
    }

    @Override
    public void
    shutdown(Ice.Current current)
    {
        current.adapter.getCommunicator().getLogger().print("Shutting down...");
        current.adapter.getCommunicator().shutdown();
    }
}
